ddspipe_participants/src/cpp/reader/rtps/CommonReader.cpp

```diff
@@ -331,6 +331,9 @@ fastrtps::rtps::ReaderAttributes CommonReader::reckon_reader_attributes_(
     if (topic.topic_qos.keyed)
     {
         att.endpoint.topicKind = eprosima::fastrtps::rtps::WITH_KEY;
+
+        // If the topic has a key, accept inline qos
+        att.expectsInlineQos = true;
```

Comment:
For consistency, maybe this should also be set in reader qos. Have you checked it is set after the registerReader call?

Reply:
The ReaderQos does not have the expectsInlineQos element; the DataReaderQos has it in the rtps layer.